init the Langchain4j plugin

aeSouid opened the pull request:
add langchaing4j dependencies
implement the textcompletion and json extrac tion fields for geemin, openai and ollama

loicmathieu commented on build.gradle:
```diff
@@ -14,129 +14,166 @@ plugins {
+        if (isBuildSnapshot) {
+            maven { url "https://s01.oss.sonatype.org/content/repositories/snapshots/" }
+        }
+        maven { url "https://maven.dremio.com/free/" }
+        maven { url 'https://maven.jumpmind.com/repo/' }
+        maven { url "https://talend-update.talend.com/nexus/content/repositories/libraries/" }
```
Should not be needed

loicmathieu commented on plugin-langchain4j-gemini/allure-results/19a24353-4ef4-43d5-930d-49c1e929cfd9-result.json:
Please exclude all allure-results directories

loicmathieu commented on plugin-langchain4j-gemini/src/main/java/io/kestra/plugin/GeminiChatMemory.java:
```diff
@@ -0,0 +1,62 @@
+        )
+    }
+)
+public class GeminiChatMemory extends AbstractChatMemory {
```
```suggestion
public class GeminiChatCompletion extends AbstractChatCompletion {
```

Also rename the other classes

loicmathieu commented on plugin-langchain4j-gemini/src/main/java/io/kestra/plugin/enums/EGeminiModel.java:
```diff
@@ -0,0 +1,18 @@
+import lombok.Getter;
+
+@Getter
+public enum EGeminiModel {
```
The name is strange, why does it start with E?
If it's because enumeration starts with e, we didn't use this kind of naming here, please switch to GeminiModel.

By the way, the list of models is provided in the OpenAI library, isn't it also provided here?

aeSouid replied:
not in GEMINI and ollama

loicmathieu commented on plugin-langchain4j-gemini/src/main/resources/application-test.yml:
```diff
@@ -0,0 +1,3 @@
+kestra:
+  gemini:
+    apikey: AIzaSyAYcZI6PrLZu3Wx03X3zGocPo5x_DXb2PA
```
I hope it's not  real API Key?

loicmathieu commented on plugin-langchain4j/src/main/java/io/kestra/plugin/AbstractChatMemory.java:
```diff
@@ -0,0 +1,106 @@
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Abstract Chat Memory Task",
+    description = "Abstract class for chat memory tasks supporting different models"
+)
```
Documentation is not needed in abstract tasks, please remove it for all abstract tasks

loicmathieu commented on plugin-langchain4j/src/main/java/io/kestra/plugin/enums/EChatType.java:
```diff
@@ -0,0 +1,12 @@
+
+@Getter
+public enum EChatType {
+    AI("Ai"), USER("User");
```
```suggestion
    AI("AI"), USER("User");
```

loicmathieu commented on plugin-langchain4j/src/main/java/io/kestra/plugin/utils/MethodUtility.java:
```diff
@@ -0,0 +1,28 @@
+import java.util.stream.Collectors;
+
+@UtilityClass
+public class MethodUtility {
```
```suggestion
public class LLMUtility {
```

It's not a utility class for methods but for LLM.
Or for ChatMessage.
Or you can just name it Utils...

loicmathieu reviewed (approved):
LGTM

aeSouid committed:
init the Langchain4j plugin
add langchaing4j dependencies
implement the textcompletion and json extrac tion fields for geemin, openai and ollama

aeSouid committed:
Langchain4j plugin
implement the Classification and json extrac tion fields for geemin, openai and ollama

aeSouid committed:
Langchain4j plugin
implement the chat completion for openai, gemini and ollama models
implement the image generation for google vertex and open ai models
